### DISCUSSION TITLE
How can I intercept events?

### DISCUSSION THREAD

**Discussion by RainingC** [fengyuanchen/viewerjs#584] [Q&A] [answered]
### Can those events be intercept???   Thanks!!!!!!

For example
![image](https://user-images.githubusercontent.com/43082163/211745531-99364488-5810-4f1a-807a-1c81f7f24f7a.png)

**For example , I click the Thumbnail's next button; I want to control the whole viewerjs to wait 3s and loading; Then switch to the next image;**

**Comment by fengyuanchen** [comment, marked as answer]
1. Some **cancelable** events (such as `show`, `hide`, `view`, etc) can be canceled by `event.preventDefault()`.
2. For your usage case, you can use the `toolbar` option to customize the click handler of the next button.
